dnf/module/repo_module_dict.py

```diff
@@ -187,11 +188,10 @@ def install(self, module_specs):
         versions, module_specs = self.get_best_versions(module_specs)
 
         for module_version, profiles, default_profiles in versions.values():
-            if module_version.repo_module.conf.locked:
-                logger.warning(module_messages[VERSION_LOCKED]
-                               .format(module_version.name,
-                                       module_version.repo_module.conf.version))
-                continue
+            to_be_installed = [profile for profile in profiles
+                               if profile not in module_version.repo_module.conf.profiles]
+            if module_version.repo_module.conf.locked and not to_be_installed:
```

Comment:
The condition may work, but it doesn't make clear sense at the first sight.
Please consider rewriting.